pkg/core/tokens/signing_key.go

```diff
@@ -25,9 +25,9 @@ func NewSigningKey() ([]byte, error) {
 	return util_rsa.FromPrivateKeyToPEMBytes(key)
 }
 
-func SigningKeyResourceKey(signingKeyPrefix string, serialNumber int, mesh string) model.ResourceKey {
-	name := fmt.Sprintf("%s-%d", signingKeyPrefix, serialNumber)
-	if serialNumber == 0 { // backwards compatibility with 1.3.x signing keys https://github.com/kumahq/kuma/issues/4006
+func SigningKeyResourceKey(signingKeyPrefix string, keyID KeyID, mesh string) model.ResourceKey {
+	name := fmt.Sprintf("%s-%s", signingKeyPrefix, keyID)
+	if keyID == KeyIDFallbackValue { // backwards compatibility with 1.3.x signing keys https://github.com/kumahq/kuma/issues/4006
```

Comment:
Maybe not in this PR, but it seems like we can remove this check now, right?

Reply:
yeah, definitely not for this PR